File: fibonacci/clojure/code-runs.clj
(ns code
  (:gen-class))

(set! *warn-on-reflection* true)
(set! *unchecked-math* :warn-on-boxed)


(defn- fibonacci ^long [^long n]
  (loop [a 0
         b 1
         i 0]
    (if (< i n)
      (recur b (+ a b) (inc i))
      a)))

(defn -main [& args]
  (let [^long u (parse-long (first args))
        ^long runs (parse-long (second args))]
    (loop [run 0]
      (let [^long r (loop [i 1
                           sum 0]
                      (if (< i u)
                        (recur (inc i) (+ sum (fibonacci i)))
                        sum))]
        (if (< run runs)
          (recur (inc run))
          (println r))))))

File: fibonacci/jvm/code_runs.java
package jvm;

public class code {

    private static int fibonacci(int n) {
        if (n <= 1) {
            return n;
        }
        int a = 0, b = 1, sum = 0;
        for (int i = 2; i <= n; i++) {
            sum = a + b;
            a = b;
            b = sum;
        }
        return b;
    }

    public static void main(String[] args) {
        var u = Integer.parseInt(args[0]);
        var runs = Integer.parseInt(args[1]);
        var r = 0;
        for (var i = 0; i < runs; i++) {
            r = 0;
            for (var j = 1; j < u; j++) {
                r += fibonacci(j);
            }
        }
        System.out.println(r);
    }
}
